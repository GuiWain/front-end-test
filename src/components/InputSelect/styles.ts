import styled, { css, DefaultTheme } from 'styled-components'

import { InputSelectProps } from './'

import { Search } from '@styled-icons/boxicons-regular/'

export type OptionsContainerProps = {
  active: boolean
}

export type OptionProps = {
  showHidde?: boolean
}

export type SelectedProps = {
  active: boolean
}

export type DivSearchProps = {
  active?: boolean
}

export type ContainerProps = {
  error?: boolean
  disabled?: boolean
  widthOptions?: number
  widthSelected?: number
} & Pick<InputSelectProps, 'searchBox'>

export const SearchIcon = styled(Search)`
  width: 2rem;
`

const containerModifiers = {
  error: (theme: DefaultTheme) => css`
    ${SelectBox} {
      border-color: ${theme.colors.errorMedium};
    }

    ${Label} {
      color: ${theme.colors.errorMedium};
    }
  `,
  disabled: (theme: DefaultTheme) => css`
    ${Label},
    ${SelectBox},
    ${OptionsContainer},
    ${Selected},
    ${Option},
    ${LabelOption} {
      cursor: not-allowed;
      color: ${theme.colors.secondaryMedium};
    }
  `,
  widthSelected: (widthSelected: number) => css`
    width: ${widthSelected + 'rem'};

    ${SelectBox}, ${Selected}, ${OptionsContainer} {
      width: ${widthSelected + 'rem'};
    }

    ${InputText} {
      width: ${widthSelected - 5 + 'rem'};
    }
  `,
  widthOptions: (widthOptions: number) => css`
    ${OptionsContainer} {
      width: ${widthOptions + 'rem'};
    }

    ${InputText} {
      width: ${widthOptions - 5 + 'rem'};
    }
  `,
  searchBox: () => css`
    ${OptionsContainer} {
      padding-top: 11rem;
      padding-bottom: 1.5rem;
    }

    ${Option} {
      :before {
        margin: 0.4rem auto;
        border-bottom: none;
      }
    }
  `
}

export const Container = styled.div<ContainerProps>`
  ${({ theme, error, disabled, widthOptions, widthSelected, searchBox }) => css`
    background: transparent;
    z-index: 999;
    width: 27rem;

    ${error && containerModifiers.error(theme)}
    ${disabled && containerModifiers.disabled(theme)}
    ${!!widthOptions && containerModifiers.widthOptions(widthOptions)}
    ${!!widthSelected && containerModifiers.widthSelected(widthSelected)}
    ${searchBox && containerModifiers.searchBox()}
  `}
`

export const SelectBox = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    width: 27rem;

    background: ${theme.colors.white} 0% 0% no-repeat padding-box;
    border: 0.1rem solid ${theme.colors.secondaryLight};
    border-radius: ${theme.border.smallRadius};
    box-sizing: content-box;

    ${Option}:first-child {
      :before {
        border-bottom: none;
      }
    }
  `}
`

const optionsContainerModifier = {
  active: (theme: DefaultTheme) => css`
    box-shadow: 0rem 0.4rem 0.8rem #00000014;

    max-height: 25rem;
    opacity: 1;
    overflow-y: scroll;
  `
}

export const OptionsContainer = styled.div.attrs({
  'data-testid': 'OptionsContainer'
})<OptionsContainerProps>`
  ${({ theme, active }) => css`
    position: absolute;
    width: 27rem;
    max-height: 0;
    padding-top: 4rem;
    padding-bottom: 1rem;

    border-radius: ${theme.border.smallRadius};
    border: 0.1rem solid ${theme.colors.secondaryLight};
    transition: all ${theme.transition.default};
    background: ${theme.colors.white} 0% 0% no-repeat padding-box;

    opacity: 0;
    overflow: hidden;

    order: 1;

    ${active && optionsContainerModifier.active(theme)}
  `}
`

const selectedModifier = {
  active: (theme: DefaultTheme) => css`
    box-shadow: 0rem 0.4rem 0.8rem #00000014;
    &:after {
      transform: rotateX(180deg);
      top: 25%;
    }
  `
}

export const Selected = styled.div.attrs({
  'data-testid': 'Selected'
})<SelectedProps>`
  ${({ theme, active }) => css`
    background: ${theme.colors.white} 0% 0% no-repeat padding-box;

    width: 27rem;
    margin: 0;
    height: 3.6rem;
    padding: 0.5rem 1.5rem;
    cursor: pointer;

    position: relative;
    display: flex;
    justify-items: flex-start;
    align-items: center;

    order: 0;

    &:after {
      content: '';
      background: url('/img/arrow-left-input-select.svg');
      background-size: contain;
      background-repeat: no-repeat;

      position: absolute;
      height: 1.2rem;
      width: 1.2rem;
      right: 1rem;
      top: 40%;

      transition: all ${theme.transition.default};
    }
    ${active && selectedModifier.active(theme)}
  `}
`

const optionModifier = {
  show: () => css`
    display: block;
  `,
  hidde: () => css`
    display: none;
  `
}

export const Option = styled.div<OptionProps>`
  ${({ theme, showHidde }) => css`
    background: ${theme.colors.white} 0% 0% no-repeat padding-box;
    margin: 0;
    padding: 0 1.5rem;

    font: normal normal ${theme.font.regular} ${theme.font.sizes.xxsmall} /
      2.4rem ${theme.font.family};
    letter-spacing: -0.006rem;
    text-align: left;
    color: ${theme.colors.secondaryMedium};

    cursor: pointer;

    :before {
      content: '';
      display: block;
      margin: 0.9rem auto;
      width: 100%;
      border-bottom: 0.1rem solid ${theme.colors.secondaryLight};
    }

    ${!showHidde && optionModifier.show()}
    ${showHidde && optionModifier.hidde()}
  `}
`

export const Input = styled.input.attrs({
  type: 'radio'
})`
  display: none;
`

export const Label = styled.p`
  ${({ theme }) => css`
    font: normal normal ${theme.font.semiBold} ${theme.font.sizes.xsmall} / 4rem
      ${theme.font.family};

    letter-spacing: -0.006rem;
    text-align: left;
    color: ${theme.colors.secondaryDark};

    margin: 0;
  `}
`

export const Span = styled.span`
  ${({ theme }) => css`
    font: normal normal ${theme.font.regular} ${theme.font.sizes.xxsmall} /
      2.4rem ${theme.font.family};

    letter-spacing: -0.006rem;
    text-align: left;
    color: ${theme.colors.secondaryMedium};
  `}
`

export const LabelOption = styled.label`
  cursor: pointer;
  width: 100%;
  display: block;
`

export const Error = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.errorMedium};
    font-size: ${theme.font.sizes.xxxsmall};
    margin: 0.5rem 0;
  `}
`

const searchBoxModifier = {
  active: (theme: DefaultTheme) => css`
    ${InputText} {
      opacity: 1;
      pointer-events: auto;

      transition: all ${theme.transition.default};
    }
  `
}

export const InputText = styled.input.attrs({
  type: 'text',
  autoFocus: true
})`
  ${({ theme }) => css`
    position: absolute;
    z-index: 9999;

    pointer-events: none;
    opacity: 0;

    background: ${theme.colors.white} 0% 0% no-repeat padding-box;
    border: 0.1rem solid ${theme.colors.secondaryLight};
    border-radius: ${theme.border.smallRadius};

    color: ${theme.colors.secondaryDark};
    font: normal normal ${theme.font.regular} ${theme.font.sizes.xxsmall} /
      2.8rem ${theme.font.family};

    padding: 0.8rem 1.6rem;
    height: 3.6rem;
    width: 22rem;
    left: 1.8rem;
    top: 2rem;
    outline: none;

    transition: all ${theme.transition.default};

    &:focus-within {
      box-shadow: 0rem 0.4rem 0.6rem #00000014;
      border: 0.1rem solid ${theme.colors.infoDark};
    }
  `}
`

export const SearchBox = styled.div<DivSearchProps>`
  ${({ theme, active }) => css`
    position: relative;
    ${active && searchBoxModifier.active(theme)}
  `}
`
