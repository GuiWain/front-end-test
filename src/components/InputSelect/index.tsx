import { useState, useEffect } from 'react'
import * as S from './styles'

export type DisplayOptions = {
  [field: string]: boolean
}

export type InputSelectProps = {
  getValues?: (field: string) => void
  options: string[]
  label?: string
  name?: string
  defaultSelect?: string
  error?: string
  disabled?: boolean
  widthOptions?: number
  widthSelected?: number
  searchBox?: boolean
}

const InputSelect = ({
  getValues,
  options,
  label,
  name,
  defaultSelect = 'Selecione...',
  error,
  disabled,
  widthOptions,
  widthSelected,
  searchBox = false
}: InputSelectProps) => {
  const [displayOption, setDisplayOption] = useState<DisplayOptions>({})
  const [selected, setSelected] = useState(defaultSelect)
  const [optionsContainer, setOptionsContainer] = useState(false)

  useEffect(() => {
    options.forEach((option) => {
      setDisplayOption((s) => ({ ...s, [option]: false }))
    })
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [])

  useEffect(() => {
    if (getValues) {
      getValues(selected)
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [selected])

  const onClickSelected = () => {
    if (disabled) {
      return
    } else setOptionsContainer(!optionsContainer)
  }

  const onClickOption = (item: string) => {
    setOptionsContainer(!optionsContainer)
    setSelected(item)
  }

  const handleInput = (value: string) => {
    const values = value.toLowerCase()

    options.forEach((option) => {
      const label = option.toLowerCase()

      if (label.indexOf(values) != -1) {
        setDisplayOption((s) => ({ ...s, [option]: false }))
      } else {
        setDisplayOption((s) => ({ ...s, [option]: true }))
      }
    })
  }

  return (
    <S.Container
      error={!!error}
      disabled={disabled}
      widthSelected={widthSelected}
      widthOptions={widthOptions}
      searchBox={searchBox}
    >
      {!!label && <S.Label>{label}</S.Label>}

      <S.SelectBox>
        <S.OptionsContainer active={optionsContainer}>
          {!!options &&
            options.map((item) => (
              <S.Option key={item} showHidde={displayOption[item]}>
                <S.Input id={item} name={name} />
                <S.LabelOption
                  htmlFor={item}
                  onClick={() => onClickOption(item)}
                >
                  {item}
                </S.LabelOption>
              </S.Option>
            ))}
        </S.OptionsContainer>
        <S.Selected onClick={onClickSelected} active={optionsContainer}>
          <S.Span>{selected}</S.Span>
        </S.Selected>

        {searchBox && (
          <S.SearchBox active={optionsContainer}>
            <S.InputText
              placeholder="&#128269; Search by"
              onChange={(e) => handleInput(e.currentTarget.value)}
            />
          </S.SearchBox>
        )}
      </S.SelectBox>
      {!!error && <S.Error>{error}</S.Error>}
    </S.Container>
  )
}

export default InputSelect
