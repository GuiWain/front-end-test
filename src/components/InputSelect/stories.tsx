import { Story, Meta } from '@storybook/react'
import InputSelect, { InputSelectProps } from '.'

import { listStates } from './mockStates'

export default {
  title: 'Inputs/InputSelect',
  component: InputSelect,
  argTypes: {
    label: {
      type: 'string'
    },
    name: {
      type: 'string'
    },
    defaultSelect: {
      type: 'string'
    },
    options: {
      type: ['string']
    },
    error: {
      type: 'string'
    },
    disabled: {
      control: {
        type: 'radio',
        options: [true, false]
      }
    },
    widthSelected: {
      type: 'number'
    },
    widthOptions: {
      type: 'number'
    },
    searchBox: {
      control: {
        type: 'radio',
        options: [true, false]
      }
    }
  }
} as Meta

export const Default: Story<InputSelectProps> = (args) => (
  <InputSelect {...args} />
)

Default.args = {
  options: listStates,
  label: 'Choose a state',
  name: 'states',
  defaultSelect: 'Select a state',
  disabled: false,
  searchBox: false
}

export const WithSearchBox: Story<InputSelectProps> = (args) => (
  <InputSelect {...args} />
)

WithSearchBox.args = {
  options: listStates,
  label: 'Choose a state',
  name: 'states',
  defaultSelect: 'Select a state',
  disabled: false,
  searchBox: true
}

export const WithError: Story<InputSelectProps> = (args) => (
  <InputSelect {...args} />
)

WithError.args = {
  options: listStates,
  label: 'Choose a state',
  name: 'states',
  defaultSelect: 'Select a state',
  error: 'Required field',
  disabled: false,
  searchBox: false
}
