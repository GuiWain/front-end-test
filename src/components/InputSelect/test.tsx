import userEvent from '@testing-library/user-event'
import { render, screen, waitFor } from 'utils/test-utils'

import InputSelect from '.'
import { listStates as mockOptions } from './mockStates'

describe('<InputSelect />', () => {
  it('should render the InputSelect', () => {
    const { container } = render(<InputSelect options={mockOptions} />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the InputSelect with label', () => {
    render(
      <InputSelect options={mockOptions} label="LabelText" name="states" />
    )

    expect(screen.getByText('LabelText')).toBeInTheDocument()
  })

  it('should render the InputSelect with error', () => {
    render(<InputSelect options={mockOptions} error="Required field" />)

    expect(screen.getByText('Required field')).toBeInTheDocument()
  })

  it('should render the InputSelect disabled', async () => {
    render(
      <InputSelect
        options={mockOptions}
        disabled
        label="LabelText"
        name="states"
      />
    )

    const input = screen.getByTestId('Selected')
    userEvent.click(input)

    await waitFor(() => {
      userEvent.click(screen.getByText('Acre - AC'))
      userEvent.click(input)
    })

    expect(screen.getByText('LabelText')).toHaveStyle({
      cursor: 'not-allowed'
    })
  })

  it('should render the InputSelect with getValues', async () => {
    const getValues = jest.fn()

    render(
      <InputSelect
        options={mockOptions}
        label="LabelText"
        name="states"
        getValues={getValues}
      />
    )

    const input = screen.getByTestId('Selected')
    userEvent.click(input)

    await waitFor(() => {
      userEvent.click(screen.getByText('Acre - AC'))
      userEvent.click(input)
    })

    expect(getValues).toHaveBeenCalled()
  })

  it('should render the InputSelect with widthSelected', () => {
    render(<InputSelect options={mockOptions} widthSelected={27} />)

    expect(screen.getByTestId('Selected')).toHaveStyle({
      width: '27rem'
    })
  })

  it('should render the InputSelect with widthOptions', () => {
    render(<InputSelect options={mockOptions} widthOptions={27} />)

    expect(screen.getByTestId('OptionsContainer')).toHaveStyle({
      width: '27rem'
    })
  })

  it('should render the InputSelect with searchBox', () => {
    render(<InputSelect options={mockOptions} searchBox={true} />)

    const inputSelect = screen.getByTestId('Selected')
    userEvent.click(inputSelect)

    const inputText = screen.getByPlaceholderText(/Search by/i)
    userEvent.type(inputText, 'Acre')

    expect(screen.queryByText(/Tocantins/i)).not.toBeVisible()
  })
})
