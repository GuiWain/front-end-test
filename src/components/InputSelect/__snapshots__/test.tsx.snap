// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputSelect /> should render the InputSelect 1`] = `
.c0 {
  background: transparent;
  z-index: 999;
  width: 27rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 27rem;
  background: #FFFFFF 0% 0% no-repeat padding-box;
  border: 0.1rem solid #CCC8C8;
  border-radius: 0.4rem;
  box-sizing: content-box;
}

.c1 .c3:first-child:before {
  border-bottom: none;
}

.c2 {
  position: absolute;
  width: 27rem;
  max-height: 0;
  padding-top: 4rem;
  padding-bottom: 1rem;
  border-radius: 0.4rem;
  border: 0.1rem solid #CCC8C8;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  background: #FFFFFF 0% 0% no-repeat padding-box;
  opacity: 0;
  overflow: hidden;
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
}

.c7 {
  background: #FFFFFF 0% 0% no-repeat padding-box;
  width: 27rem;
  margin: 0;
  height: 3.6rem;
  padding: 0.5rem 1.5rem;
  cursor: pointer;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  justify-items: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c7:after {
  content: '';
  background: url('/img/arrow-left-input-select.svg');
  background-size: contain;
  background-repeat: no-repeat;
  position: absolute;
  height: 1.2rem;
  width: 1.2rem;
  right: 1rem;
  top: 40%;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
}

.c4 {
  background: #FFFFFF 0% 0% no-repeat padding-box;
  margin: 0;
  padding: 0 1.5rem;
  font: normal normal 400 1.4rem / 2.4rem 'Open Sans','Roboto','Helvetica Neue',sans-serif;
  -webkit-letter-spacing: -0.006rem;
  -moz-letter-spacing: -0.006rem;
  -ms-letter-spacing: -0.006rem;
  letter-spacing: -0.006rem;
  text-align: left;
  color: #848080;
  cursor: pointer;
  display: block;
}

.c4:before {
  content: '';
  display: block;
  margin: 0.9rem auto;
  width: 100%;
  border-bottom: 0.1rem solid #CCC8C8;
}

.c5 {
  display: none;
}

.c8 {
  font: normal normal 400 1.4rem / 2.4rem 'Open Sans','Roboto','Helvetica Neue',sans-serif;
  -webkit-letter-spacing: -0.006rem;
  -moz-letter-spacing: -0.006rem;
  -ms-letter-spacing: -0.006rem;
  letter-spacing: -0.006rem;
  text-align: left;
  color: #848080;
}

.c6 {
  cursor: pointer;
  width: 100%;
  display: block;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
      data-testid="OptionsContainer"
    >
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Acre - AC"
          type="radio"
        />
        <label
          class="c6"
          for="Acre - AC"
        >
          Acre - AC
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Alagoas - AL"
          type="radio"
        />
        <label
          class="c6"
          for="Alagoas - AL"
        >
          Alagoas - AL
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Amapá - AP"
          type="radio"
        />
        <label
          class="c6"
          for="Amapá - AP"
        >
          Amapá - AP
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Amazonas - AM"
          type="radio"
        />
        <label
          class="c6"
          for="Amazonas - AM"
        >
          Amazonas - AM
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Bahia - BA"
          type="radio"
        />
        <label
          class="c6"
          for="Bahia - BA"
        >
          Bahia - BA
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Ceará - CE"
          type="radio"
        />
        <label
          class="c6"
          for="Ceará - CE"
        >
          Ceará - CE
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Distrito federal - DF"
          type="radio"
        />
        <label
          class="c6"
          for="Distrito federal - DF"
        >
          Distrito federal - DF
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Espiríto Santo - ES"
          type="radio"
        />
        <label
          class="c6"
          for="Espiríto Santo - ES"
        >
          Espiríto Santo - ES
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Goiás - GO"
          type="radio"
        />
        <label
          class="c6"
          for="Goiás - GO"
        >
          Goiás - GO
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Maranhão - MA"
          type="radio"
        />
        <label
          class="c6"
          for="Maranhão - MA"
        >
          Maranhão - MA
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Mato Grosso - MT"
          type="radio"
        />
        <label
          class="c6"
          for="Mato Grosso - MT"
        >
          Mato Grosso - MT
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Mato Grosso do Sul - MS"
          type="radio"
        />
        <label
          class="c6"
          for="Mato Grosso do Sul - MS"
        >
          Mato Grosso do Sul - MS
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Minas Gerais - MG"
          type="radio"
        />
        <label
          class="c6"
          for="Minas Gerais - MG"
        >
          Minas Gerais - MG
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Pará - PA"
          type="radio"
        />
        <label
          class="c6"
          for="Pará - PA"
        >
          Pará - PA
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Paraíba - PB"
          type="radio"
        />
        <label
          class="c6"
          for="Paraíba - PB"
        >
          Paraíba - PB
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Paraná - PR"
          type="radio"
        />
        <label
          class="c6"
          for="Paraná - PR"
        >
          Paraná - PR
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Pernambuco - PE"
          type="radio"
        />
        <label
          class="c6"
          for="Pernambuco - PE"
        >
          Pernambuco - PE
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Piauí - PI"
          type="radio"
        />
        <label
          class="c6"
          for="Piauí - PI"
        >
          Piauí - PI
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Rio de Janeiro - RJ"
          type="radio"
        />
        <label
          class="c6"
          for="Rio de Janeiro - RJ"
        >
          Rio de Janeiro - RJ
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Rio Grande do Norte - RN"
          type="radio"
        />
        <label
          class="c6"
          for="Rio Grande do Norte - RN"
        >
          Rio Grande do Norte - RN
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Rio Grande do Sul - RS"
          type="radio"
        />
        <label
          class="c6"
          for="Rio Grande do Sul - RS"
        >
          Rio Grande do Sul - RS
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Rondônia - RO"
          type="radio"
        />
        <label
          class="c6"
          for="Rondônia - RO"
        >
          Rondônia - RO
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Roraima - RR"
          type="radio"
        />
        <label
          class="c6"
          for="Roraima - RR"
        >
          Roraima - RR
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Santa Catarina - SC"
          type="radio"
        />
        <label
          class="c6"
          for="Santa Catarina - SC"
        >
          Santa Catarina - SC
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="São Paulo - SP"
          type="radio"
        />
        <label
          class="c6"
          for="São Paulo - SP"
        >
          São Paulo - SP
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Sergipe - SE"
          type="radio"
        />
        <label
          class="c6"
          for="Sergipe - SE"
        >
          Sergipe - SE
        </label>
      </div>
      <div
        class="c3 c4"
      >
        <input
          class="c5"
          id="Tocantins - TO"
          type="radio"
        />
        <label
          class="c6"
          for="Tocantins - TO"
        >
          Tocantins - TO
        </label>
      </div>
    </div>
    <div
      class="c7"
      data-testid="Selected"
    >
      <span
        class="c8"
      >
        Selecione...
      </span>
    </div>
  </div>
</div>
`;
